src/Root.ts

```diff
@@ -256,31 +269,60 @@ export class RootAPI implements Types.RootAPI {
     private _win: Types.GetWindow;
     private _initTimer: number | undefined;
     private _autoRoot: Types.RootProps | undefined;
-    private _autoRootInstance: Root | undefined;
+    private _autoRootWaiting = false;
     private _roots: Record<string, Types.Root> = {};
+    private _forceDummy = false;
     rootById: { [id: string]: Types.Root } = {};
     eventTarget: EventTarget;
 
     constructor(tabster: Types.TabsterCore, autoRoot?: Types.RootProps) {
         this._tabster = tabster;
         this._win = tabster.getWindow;
         this._initTimer = this._win().setTimeout(this._init, 0);
-        this._autoRoot = autoRoot;
+        if (autoRoot) {
+            this._autoRoot = autoRoot;
+            this._autoRootCreate();
+        }
         this.eventTarget = createEventTarget(this._win);
     }
 
     private _init = (): void => {
         this._initTimer = undefined;
     };
 
+    private _autoRootCreate = (): Types.Root | undefined => {
+        const doc = this._win().document;
+        const body = doc.body;
+
+        if (body) {
+            this._autoRootUnwait(doc);
+
+            const props = this._autoRoot;
+
+            if (props) {
+                setTabsterAttribute(body, { root: props }, true);
+                return getTabsterOnElement(this._tabster, body)?.root;
+            }
+        } else if (!this._autoRootWaiting) {
+            this._autoRootWaiting = true;
+            doc.addEventListener("readystatechange", this._autoRootCreate);
+        }
+
+        return undefined;
+    };
+
+    private _autoRootUnwait(doc: Document): void {
+        if (this._autoRootWaiting) {
```

Comment:
you don't need this condition right? `removeEventListener` does not throw when it cannot find the event listener, and you can just always set the flag to falst